Align the Swift implementation with the charter

kylef opened the pull request:
The main thing that is left is figuring out the structure for the validators of an InputProperty which I'm going to work on next.

Post merge tasks:
- [ ] Update https://github.com/mikekelly/hal_specification/wiki/Libraries

kylef committed:
Build initial Resource and Transition structures

kylef committed:
[README] Contain example usage

kylef committed:
[Transition] Provide builder method for adding attributes without value

kylef committed:
Include builder pattern to construct resources and transitions

kylef committed:
[Transition] No longer conform to Equality

Unfortunately this proves difficult when we have properties of AnyObject
type

kylef committed:
[Transition] Add attributes and parameters

kylef committed:
Initial Xcode project

kylef committed:
Move existing source into project

kylef committed:
[travis] Enable CI

kylef committed:
[Resource] Add links to other resources

kylef committed:
[Resource] Add metadata

kylef committed:
Rename resource to representor

kylef committed:
[Project] Set deployment target to 10.9

kylef committed:
Rename project to Representor

kylef committed:
[README] Update repository location

segiddins commented on .travis.yml:
```diff
@@ -0,0 +1,4 @@
+language: objective-c
```
this is swift, Kyle :)

kylef replied:
Just waiting on https://github.com/travis-ci/travis-build/pull/323 ;)

zdne replied:
:)

kylef committed:
[Representor] Support multiple embedded representors

kylef committed:
Add support for conversion to/from HAL

kylef committed:
Add support for conversion to/from Siren

kylef committed:
Turn InputProperty into a generic type

This allows type safety in the transition builder, the following will
fail to compile due to different types:

    builder.addAttribute("name", "Kyle", 5)

kylef committed:
[HAL Tests] Simplify testing logic

kylef committed:
Add conformance to Equatable

kylef committed:
[Tests] Refactor HAL to use external fixtures

kylef committed:
[Siren] Support embedded documents

kylef committed:
[Siren] Convert embedded representors during `asSiren`

kylef committed:
[README] Mention the HAL and Siren adapters

kylef committed:
[Tests] Use fixtures for converting to HAL

kylef committed:
[Podspec] Initial creation

kylef committed:
[Tests] Include next link in fixtures

kylef commented:
I've added the next links to the fixtures (@fosrias).

I'd like if we can get this merged so we can start creating separate pull requests instead of adding work onto this one. I've created #2 to track the outstanding work of improving the Siren adapter.

zdne commented:
@kylef is there any work pending or is this good to merge?

kylef commented:
@zdne Let's merge this :shipit:.

kylef committed:
[Project] Switch to two spaces indentation

kylef committed:
[Builder] Document methods

kylef committed:
[Representor] Conform to Hashable

kylef committed:
[Transition] Conform to Hashable

kylef committed:
[Representor] Add initialiser from NSHTTPURLResponse

kylef committed:
[README] Document NSHTTPURLResponse

kylef committed:
[README] Updated description

kylef committed:
[Podspec] Add Response subspec

kylef committed:
[Project] Convert to universal framework

kylef committed:
[travis] Test on iOS and OS X

zdne commented:
Merged.Although I would still add following to .gitignore:

```
Representor.xcodeproj/project.xcworkspace/xcshareddata/Representor.xccheckout
Representor.xcodeproj/project.xcworkspace/xcuserdata/zdenek.xcuserdatad/UserInterfaceState.xcuserstate
Representor.xcodeproj/xcuserdata/zdenek.xcuserdatad/xcschemes/xcschememanagement.plist
```

kylef commented:
@zdne, I'll sort a git ignore out tomorrow
